### PR TITLE
Trivyignore

### PR DESCRIPTION
**Pull request by Bslabe123**
# Description

Added a field to SecurityScanner to allow the passing of trivyignore file contents.

# Context

This feature is needed to prevent the go-restful CVE from recurring weekly even with a trivyignore.
This solution is not framework agnostic, this could be fixed by parsing the trivyignore and passing in a list of CVE codes to ignore.

# Checklist:

- [ ] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [ ] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [ ] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [ ] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

BOT NOTES: 
resolves https://github.com/solo-io/solo-projects/issues/3907

### CODE REVIEW

File: docs/cmd/generate_docs.go [hunk 1]
```diff
@@ -306,6 +306,8 @@ func scanGlooImages(ctx context.Context) error {
 			log.Fatalf("Invalid constraint version: %s", minVersionToScan)
 		}
 	}
+
+	a, err := os.ReadFile(".trivyignore")
```
Comment (nfuden) [inline docs/cmd/generate_docs.go:310]: why is this called a?

Comment (nfuden) [reply docs/cmd/generate_docs.go:310]: also err is unchecked... should we just chomp it and have empty string be expected?

Comment (Bslabe123) [reply docs/cmd/generate_docs.go:310]: No reason, just a placeholder I forgot to remove

Comment (Bslabe123) [reply docs/cmd/generate_docs.go]: Yep that was the plan, if it errors out it just returns an empty trying which is totally acceptable to write to the new trivyignore

File: go.mod [hunk 2]
```diff
@@ -81,7 +81,7 @@ require (
 	github.com/smartystreets/assertions v1.2.0 // indirect
 	github.com/solo-io/anyvendor v0.0.4 // indirect
 	github.com/solo-io/go-list-licenses v0.1.0
-	github.com/solo-io/go-utils v0.22.0
+	github.com/solo-io/go-utils v0.22.1-0.20220802185301-482c81bdf485
```
Comment (jackstine) [inline go.mod:84]: why is not a release of `go-utis`?

**Pull request closed without merging**